pkg/controller/registry/resolver/steps.go

```diff
@@ -89,8 +89,8 @@ func NewStepResourceFromObject(obj runtime.Object, catalogSourceName, catalogSou
 func NewSubscriptionStepResource(namespace string, info OperatorSourceInfo) (v1alpha1.StepResource, error) {
 	return NewStepResourceFromObject(&v1alpha1.Subscription{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: namespace,
-			Name:      strings.Join([]string{info.Package, info.Channel, info.Catalog.Name, info.Catalog.Namespace}, "-"),
+			Namespace:    namespace,
+			GenerateName: strings.ToLower(fmt.Sprintf("%s-%s-%s-%s-", info.Package, info.Channel, info.Catalog.Name, info.Catalog.Namespace)),
```

Comment:
I think we also need to limit the length of this field to adhere to the limitations of the `Name` field. From [the metav1.Object source](https://github.com/kubernetes/kubernetes/blob/0b4a30bb25cfcf7c0895361dea2a3d465c75bd0e/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L136), `GenerateName`:

> If this field is used, the name returned to the client will be different than the name passed. This value will also be combined with a unique suffix. The provided value has the same validation rules as the Name field, and may be truncated by the length of the suffix required to make the value unique on the server.

Following the [naming rules](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names) gives us a max length of 253 characters, however to be safe, I suggest we bound the length by the max value of a [DNS label](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names), 63 characters. This lets us account for anything that may use Subscription names for resources that are bound by the lower limit.